bindings/python/CMakeLists.txt

```diff
@@ -1,39 +1,42 @@
 set(CMAKE_SWIG_FLAGS "-Wextra;-module;iDynTree;-threads")
 set(IDYNTREE_USES_PYTHON_VERSION "" CACHE STRING "Specify python version to use" )
 
-find_package(PythonInterp ${IDYNTREE_USES_PYTHON_VERSION} REQUIRED)
-set(IDYNTREE_USES_PYTHON_VERSION_ARGS)
-  if(NOT IDYNTREE_USES_PYTHON_VERSION)
-     set (IDYNTREE_USES_PYTHON_VERSION ${PYTHON_VERSION_STRING})
-  endif()
+find_package(Python3 ${IDYNTREE_USES_PYTHON_VERSION} COMPONENTS Interpreter Development REQUIRED)
```

Comment:
As long as this is targeting `devel` (i.e. iDynTree 2), it is ok to just support Python3 . In this context, what is the value of `IDYNTREE_USES_PYTHON_VERSION` ? Permit to find for different Python 3 version? I am not sure if `FindPython3` works in this way.

Reply:
If we can ignore Python2 support, I'll remove `IDYNTREE_USES_PYTHON_VERSION` entirely. If you want to find a specific version, you can use `find_package(Python3 3.6 EXACT [...])`, however this is not the case since we just need any Python3. Done in e6b6f50.